alf/algorithms/agent.py

```diff
@@ -127,6 +127,7 @@ def __init__(self,
         rl_algorithm = rl_algorithm_cls(
             observation_spec=rl_observation_spec,
             action_spec=action_spec,
+            config=config,  # set use_rollout_state
```

Comment:
The @use_rollout_property.setter is supposed to handle this. Apparently, it does not have effect because rl_algorithm is not set as the property of self before the base constructor is called.

Reply:
So what's your suggestion?

Reply:
> The @use_rollout_property.setter is supposed to handle this. Apparently, it does not have effect because rl_algorithm is not set as the property of self before the base constructor is called.

Seems that pytorch won't allow setting a child module before the parent's constructor is called.

Reply:
Previously the setter is called by policy trainer. So there is no problem. Do you think we should remove the setter now?

Reply:
Now I moved the setter to the end of agent's init.